src/reverselock.h

```diff
@@ -13,7 +13,7 @@ class reverse_lock
 {
 public:
 
-    explicit reverse_lock(Lock& lock) : lock(lock) {
+    explicit reverse_lock(Lock& _lock) : lock(_lock) {
```

Comment:
Having the squashme commit clearly marked enables us to discuss the change needed.